File: leetcode/lc-us/cpp/2461.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    long long maximumSubarraySum(vector<int> &nums, int k) {
        unordered_map<int, int> cnt;
        int n = (int) nums.size();
        long long res = 0, sum = 0;
        for (int j = 0, i = 0; i < n; i++) {
            sum += nums[i];
            cnt[nums[i]]++;
            if (i - j + 1 == k) {
                if (cnt.size() == k) res = max(res, sum);
                sum -= nums[j];
                cnt[nums[j]]--;
                if (cnt[nums[j]] == 0) cnt.erase(nums[j]);
                j++;
            }
        }
        return res;
    }
};

File: leetcode/lc-us/py/2461.py
from collections import Counter

from typing import List


class Solution:
    def maximumSubarraySum(self, nums: List[int], k: int) -> int:
        s, res = sum(nums[:k - 1]), 0
        cnt = Counter(nums[0:k - 1])
        for come, go in zip(nums[k - 1:], nums):
            cnt[come] += 1
            s += come
            if len(cnt) == k:
                res = max(res, s)
            s -= go
            cnt[go] -= 1
            if not cnt[go]:
                del cnt[go]
        return res
